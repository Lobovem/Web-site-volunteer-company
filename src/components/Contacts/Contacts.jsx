/* eslint-disable jsx-a11y/iframe-has-title */
import s from './Contacts.module.scss';

export const Contacts = () => {
  return (
    <div className={s.contacts}>
      <div className={s.contacts__mapWrap}>
        <iframe
          className={s.contacts__map}
          src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2563.803122326351!2d36.21800665182715!3d50.015045226420746!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x4127a1234e73cbdb%3A0xcdfc0bda6e6ba6bc!2z0YPQuy4g0JrQvtGB0LzQuNGH0LXRgdC60LDRjywgMjEsINCl0LDRgNGM0LrQvtCyLCDQpdCw0YDRjNC60L7QstGB0LrQsNGPINC-0LHQu9Cw0YHRgtGMLCA2MTAwMA!5e0!3m2!1sru!2sua!4v1677322704623!5m2!1sru!2sua"
          frameBorder="0"
          loading="lazy"
          data-testid="map-iframe"
          ferrerpolicy="no-referrer-when-downgrade"
        ></iframe>
      </div>

      <div className={s.contacts__list}>
        <div className={s.contacts__item}>
          <p className={s.contacts__itemTitle}>Графік роботи:</p>
          <p className={s.contacts__itemDesc}>Пн-Пт - 09:00-16:00</p>
        </div>

        <div className={s.contacts__item}>
          <p className={s.contacts__itemTitle}>Контактні телефони:</p>
          <p className={s.contacts__itemDesc}>+380 (63)-049-40-02</p>
          <p className={s.contacts__itemDesc}>+380 (67)-159-73-73</p>
        </div>

        <div className={s.contacts__item}>
          <p className={s.contacts__itemTitle}>Адреса:</p>
          <p className={s.contacts__itemDesc}>м. Харків, вул.Космічна, буд. 21, оф. 529</p>
        </div>
      </div>
    </div>
  );
};
