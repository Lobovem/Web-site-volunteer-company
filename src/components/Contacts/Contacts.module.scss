@import './src/styles/MixinsMedia.module.scss';
@import './src/styles/Variables.module.scss';
@import './src/styles/MixinsLayout.module.scss';
@import './src/styles/MixinsKit.module.scss';

.contacts {

  &__mapWrap {
    overflow: hidden;
    padding-bottom: 56.25%;
    position: relative;
    height: 0;
    border-radius: 12px;
  }

  &__map {
    left: 0;
    top: 0;
    height: 100%;
    width: 100%;
    position: absolute;
  }

  &__list {
    display: flex;
    flex-direction: column;
    gap: 20px;

    @include lg-min {
      flex-direction: row;

    }

    @include xl-min {
      margin-left: -80px;
      margin-right: -80px;
    }
  }

  &__item {
    background: $color-white;
    box-shadow: 0px 4px 15px rgba(2, 108, 167, 0.08);
    border-radius: 14px;
    padding: 40px;

    @include lg-min {
      flex: 1;
    }

    @include xl-min {
      padding: 80px;

    }
  }

  &__itemTitle {
    margin: 0;
    padding: 0;
    font-style: normal;
    font-weight: $font-weight-600;
    font-size: $font-size-16;
    line-height: 16px;
    margin-bottom: 10px;

    @include xl-min {
      font-size: $font-size-24;
      line-height: 24px;
    }
  }

  &__itemDesc {
    padding-bottom: 6px;
    margin: 0;

    @include xl-min {
      font-size: $font-size-20;
    }
  }
}