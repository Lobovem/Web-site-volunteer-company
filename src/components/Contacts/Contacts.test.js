import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Contacts } from './Contacts';

describe('Contacts component', () => {
  test('renders Contacts component', () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const component = render(
      <Router>
        <Contacts />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

  test('renders contacts information correctly', () => {
    // Render the Contacts component
    render(<Contacts />);

    // Assert that the contact information is rendered correctly
    expect(screen.getByText('Графік роботи:')).toBeInTheDocument();
    expect(screen.getByText('Пн-Пт - 09:00-16:00')).toBeInTheDocument();
    expect(screen.getByText('Контактні телефони:')).toBeInTheDocument();
    expect(screen.getByText('Адреса:')).toBeInTheDocument();
    expect(screen.getByText('м. Харків, вул.Космічна, буд. 21, оф. 529')).toBeInTheDocument();
  });

  test('renders and search iframe element with correct attributes', () => {
    render(<Contacts />);
    const iframeElement = screen.getByTestId('map-iframe');
    expect(iframeElement).toBeInTheDocument();
  });
});
