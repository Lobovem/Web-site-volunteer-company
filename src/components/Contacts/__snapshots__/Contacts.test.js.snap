// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts component renders Contacts component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="contacts"
      >
        <div
          class="contacts__mapWrap"
        >
          <iframe
            class="contacts__map"
            data-testid="map-iframe"
            ferrerpolicy="no-referrer-when-downgrade"
            frameborder="0"
            loading="lazy"
            src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2563.803122326351!2d36.21800665182715!3d50.015045226420746!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x4127a1234e73cbdb%3A0xcdfc0bda6e6ba6bc!2z0YPQuy4g0JrQvtGB0LzQuNGH0LXRgdC60LDRjywgMjEsINCl0LDRgNGM0LrQvtCyLCDQpdCw0YDRjNC60L7QstGB0LrQsNGPINC-0LHQu9Cw0YHRgtGMLCA2MTAwMA!5e0!3m2!1sru!2sua!4v1677322704623!5m2!1sru!2sua"
          />
        </div>
        <div
          class="contacts__list"
        >
          <div
            class="contacts__item"
          >
            <p
              class="contacts__itemTitle"
            >
              Графік роботи:
            </p>
            <p
              class="contacts__itemDesc"
            >
              Пн-Пт - 09:00-16:00
            </p>
          </div>
          <div
            class="contacts__item"
          >
            <p
              class="contacts__itemTitle"
            >
              Контактні телефони:
            </p>
            <p
              class="contacts__itemDesc"
            >
              +380 (63)-049-40-02
            </p>
            <p
              class="contacts__itemDesc"
            >
              +380 (67)-159-73-73
            </p>
          </div>
          <div
            class="contacts__item"
          >
            <p
              class="contacts__itemTitle"
            >
              Адреса:
            </p>
            <p
              class="contacts__itemDesc"
            >
              м. Харків, вул.Космічна, буд. 21, оф. 529
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="contacts"
    >
      <div
        class="contacts__mapWrap"
      >
        <iframe
          class="contacts__map"
          data-testid="map-iframe"
          ferrerpolicy="no-referrer-when-downgrade"
          frameborder="0"
          loading="lazy"
          src="https://www.google.com/maps/embed?pb=!1m18!1m12!1m3!1d2563.803122326351!2d36.21800665182715!3d50.015045226420746!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x4127a1234e73cbdb%3A0xcdfc0bda6e6ba6bc!2z0YPQuy4g0JrQvtGB0LzQuNGH0LXRgdC60LDRjywgMjEsINCl0LDRgNGM0LrQvtCyLCDQpdCw0YDRjNC60L7QstGB0LrQsNGPINC-0LHQu9Cw0YHRgtGMLCA2MTAwMA!5e0!3m2!1sru!2sua!4v1677322704623!5m2!1sru!2sua"
        />
      </div>
      <div
        class="contacts__list"
      >
        <div
          class="contacts__item"
        >
          <p
            class="contacts__itemTitle"
          >
            Графік роботи:
          </p>
          <p
            class="contacts__itemDesc"
          >
            Пн-Пт - 09:00-16:00
          </p>
        </div>
        <div
          class="contacts__item"
        >
          <p
            class="contacts__itemTitle"
          >
            Контактні телефони:
          </p>
          <p
            class="contacts__itemDesc"
          >
            +380 (63)-049-40-02
          </p>
          <p
            class="contacts__itemDesc"
          >
            +380 (67)-159-73-73
          </p>
        </div>
        <div
          class="contacts__item"
        >
          <p
            class="contacts__itemTitle"
          >
            Адреса:
          </p>
          <p
            class="contacts__itemDesc"
          >
            м. Харків, вул.Космічна, буд. 21, оф. 529
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
